api/src/main/java/org/apache/iceberg/encryption/EncryptingFileIO.java

```diff
@@ -49,13 +50,17 @@ public static EncryptingFileIO combine(FileIO io, EncryptionManager em) {
   }
 
   private final FileIO io;
-  private final EncryptionManager em;
+  private EncryptionManager em;
 
   EncryptingFileIO(FileIO io, EncryptionManager em) {
     this.io = io;
     this.em = em;
   }
 
+  public void setEncryptionManager(EncryptionManager em) {
+    this.em = em;
+  }
```

Comment:
Not ideal. Hacky way of fixing the issue of the IO tracker invoking `ops.io()` before a commit.